### PR TITLE
Fix cp promise returns by converting the exec callbacks into promises

### PR DESCRIPTION
**Pull request by joeferner**
Addresses issue https://github.com/kubernetes-client/javascript/issues/1767

Wraps the exec in a promise properly such that awaiting the promise will only return after the file is copied

### CODE REVIEW

**Comment by brendandburns** [comment]
Thanks for the PR! Can you add a unit test demonstrating the problem as well so that we can be sure this fix works (and that it stays fixed)

**Comment by joeferner** [comment]
enhanced unit tests for both `cpFromPod` and `cpToPod` to test both the promises and that the data send/received was correct. Also deleted the temporary file created when running `cpToPod`

File: src/cp_test.ts [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
 import { Exec } from './exec';
 import { Cp } from './cp';
-import { WebSocketHandler, WebSocketInterface } from './web-socket-handler';
+import { BinaryHandler, WebSocketHandler, WebSocketInterface } from './web-socket-handler';
+import { V1Status } from './api';
+import { randomUUID } from 'crypto';
+import { sleep } from './util';
 
 describe('Cp', () => {
+    let tmpDir: string | undefined;
```
Comment (mstruebing) [inline src/cp_test.ts:19]: If you initialize `tmpDir` here with an empty string you don't need to check in every test for it, you could even initialize it with a proper dir:

```suggestion
    let tmpDir: string = `${tmpdir()}/${randomUUID()}`;
```

**Review by mstruebing — COMMENTED** [review]
Only one very minor comment, otherwise this looks like a solid PR to me.

**Pull request merged**